### PR TITLE
feat: use new get-starknet

### LINKED ISSUE #10: `enable()` is a legacy and deprecated method
https://docs.metamask.io/guide/ethereum-provider.html#legacy-methods

image

### PR DESCRIPTION
**Pull request by janek26**
closes #10

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -45,12 +45,17 @@ yarn add @argent/get-starknet starknet
...
-const starknet = getStarknet()
-// May throw when no extension is detected, otherwise shows a modal prompting the user to download Argent X.
-const [userWalletContractAddress] = await starknet.enable({ showModal: true })
+// Let the user pick a wallet (on button click)
+const starknet = connect()
+
+// or try to connect to an approved wallet silently (on mount probably)
+const starknet = connect({ showList: false })
+
+if(!starknet) {
```
Comment (delaaxe) [inline Readme.md:56]:
```suggestion
if (!starknet) {
```

File: packages/dapp/src/components/TokenDapp.tsx [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const TokenDapp: FC = () => {
   }, [transactionStatus, lastTransactionHash])
 
   const network = networkId()
+  console.log("network", network)
```
Comment (delaaxe) [inline packages/dapp/src/components/TokenDapp.tsx:42]: console

File: packages/dapp/src/services/wallet.service.ts [hunk 3]
```diff
@@ -30,7 +44,9 @@ export const networkId = (): Network | undefined => {
 
 export const addToken = async (address: string): Promise<void> => {
   const starknet = getStarknet()
-  await starknet.enable()
+  if (!starknet || starknet.isConnected === false) {
```
Comment (delaaxe) [inline packages/dapp/src/services/wallet.service.ts:47]: shouldn't this be?
```suggestion
  if (!starknet?.isConnected) {
```

File: packages/dapp/src/services/wallet.service.ts [hunk 4]
```diff
@@ -88,15 +106,30 @@ export const signMessage = async (message: string) => {
...
 }
 
-export const waitForTransaction = async (hash: string) =>
-  await getStarknet().provider.waitForTransaction(hash)
+export const waitForTransaction = async (hash: string) => {
+  const starknet = getStarknet()
+  if (!starknet || starknet.isConnected === false) {
+    return
+  }
+  return starknet.provider.waitForTransaction(hash)
```
Comment (delaaxe) [inline packages/dapp/src/services/wallet.service.ts:114]: missing await but actually it doesn't matter

Comment (janek26) [reply packages/dapp/src/services/wallet.service.ts:114]: right, I think having await in return statements is unnecessary

File: packages/dapp/src/pages/index.tsx [hunk 5]
```diff
@@ -1,46 +1,45 @@
...
-    addWalletChangeListener((accounts) => {
-      if (accounts.length > 0) {
-        setAddress(accounts[0])
-      } else {
-        setAddress("")
-        setIsConnected(false)
-      }
-    })
-  }, [])
+  const [isConnected, setConnected] = useState<boolean>(false)
```
Comment (delaaxe) [inline packages/dapp/src/pages/index.tsx:19]:
```suggestion
  const [isConnected, setConnected] = useState(false)
```

File: packages/dapp/src/services/wallet.service.ts [hunk 6]
```diff
@@ -1,23 +1,37 @@
-import { getStarknet } from "@argent/get-starknet"
+import { connect, getStarknet } from "@argent/get-starknet"
 import { shortString } from "starknet"
 
 import { Network } from "./token.service"
 
-export const isWalletConnected = (): boolean => !!getStarknet()?.isConnected
+export const silentConnectWallet = async () => {
+  const windowStarknetRef = await connect({ showList: false })
```
Comment (delaaxe) [inline packages/dapp/src/services/wallet.service.ts:7]: not a ref in the react sense

```suggestion
  const windowStarknet = await connect({ showList: false })
```

File: packages/dapp/src/services/wallet.service.ts (hunk 3 above)
Comment (delaaxe) [reply packages/dapp/src/services/wallet.service.ts:47]: so this doesn't work typescript-wise?

File: packages/dapp/src/services/wallet.service.ts [hunk 7]
```diff
@@ -88,15 +106,30 @@ export const signMessage = async (message: string) => {
...
 
 export const addWalletChangeListener = async (
   handleEvent: (accounts: string[]) => void,
 ) => {
-  getStarknet().on("accountsChanged", handleEvent)
+  const starknet = getStarknet()
+  if (!starknet || starknet.isConnected === false) {
+    return
+  }
+  starknet.on?.("accountsChanged", handleEvent)
```
Comment (delaaxe) [inline packages/dapp/src/services/wallet.service.ts:124]: are on/off methods really nullable?

**Review by delaaxe — APPROVED** [review]

**Comment by delaaxe** [comment]
closes #10

**Pull request merged**